appservice/src/FileUtilities.ts

```diff
@@ -40,30 +27,67 @@ export async function zipFile(filePath: string): Promise<string> {
     return zipFilePath;
 }
 
-export async function zipDirectory(folderPath: string, globPattern: string = '**/*', ignorePattern?: string | string[]): Promise<string> {
+async function zipDirectoryInternal(folderPath: string, addFiles: (zipper: archiver.Archiver) => Promise<void>): Promise<string> {
     if (!folderPath.endsWith(path.sep)) {
         folderPath += path.sep;
     }
 
     const zipFilePath: string = path.join(os.tmpdir(), `${randomFileName()}.zip`);
-    await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
-        const zipOutput: fs.WriteStream = fs.createWriteStream(zipFilePath);
+    await new Promise(async (resolve: () => void, reject: (err: Error) => void): Promise<void> => {
+        const zipOutput: fse.WriteStream = fse.createWriteStream(zipFilePath);
         zipOutput.on('close', resolve);
 
         const zipper: archiver.Archiver = archiver('zip', { zlib: { level: 9 } });
         zipper.on('error', reject);
+        await addFiles(zipper);
         zipper.pipe(zipOutput);
-        zipper.glob(globPattern, {
-            cwd: folderPath,
-            dot: true,
-            ignore: ignorePattern
-        });
         void zipper.finalize();
     });
 
     return zipFilePath;
 }
 
+/**
+ * Zips directory using glob filtering
+ */
+export async function zipDirectoryGlob(folderPath: string, globPattern: string = '**/*', ignorePattern?: string | string[]): Promise<string> {
+    return await zipDirectoryInternal(folderPath, async (z) => {
+        await addFilesByGlob(z, folderPath, globPattern, ignorePattern);
+    });
+}
+
+/**
+ * Zips directory using gitignore filtering
+ */
+export async function zipDirectoryGitignore(folderPath: string, gitignoreName: string): Promise<string> {
```

Comment:
Just a question, but is it okay to not have an option to respect both the userSettings and the `.funcignore`?  Should it always be one or the other?

Reply:
It's definitely an option to support both, but I decided it wasn't worth it (willing to discuss if you disagree). For the following reasons:
1. There would be confusion between which takes priority
1. I'd prefer to have one consistent way to do this and the func cli supports ".funcignore"
1. .gitignore patterns are much easier than glob patterns IMO
1. In functions we never prompt the user for these settings (like you do for app service) so I'm guessing they have low usage anyways (users would've had to discover them themselves)

Reply:
Also both you and Matt seemed to imply you preferred just one consistent option in your comments here: https://github.com/Microsoft/vscode-azurefunctions/issues/418